neptune/new/exceptions.py

```diff
@@ -45,6 +45,25 @@ class MetadataInconsistency(NeptuneException):
     pass
 
 
+class MissingAttributeException(NeptuneException):
```

Comment:
@aniezurawski suggested that we should raise KeyError or KeyError like error as we are mimicking dict interface.

Reply:
Good point.

I've kept `MissingAttributeException`, but inherited addtionally from:
* `KeyError` - because of dict like data access
* `AttributeError` for backward compatibility

And added proper tests for catching `MissingAttributeException` as every mentioned exception type.